webapp/channels/src/actions/burn_on_read_deletion.ts

```diff
@@ -35,11 +35,21 @@ export function burnPostNow(postId: string): ActionFuncAsync<boolean> {
  * Also used when receiving WebSocket expiration event
  */
 export function handlePostExpired(postId: string): ActionFuncAsync<boolean> {
-    return async (dispatch) => {
+    return async (dispatch, getState) => {
+        // Get the full post object from state so we can properly remove it from all reducers
+        const state = getState();
+        const post = state.entities.posts.posts[postId];
```

Comment:
It would be better to call the `getPost` selector to keep this logic isolated in one, single place.